File: android/app/src/main/java/voidawt/datatransfer/StringSelection.java
package voidawt.datatransfer;

import java.io.IOException;

public class StringSelection implements Transferable {
    private final String data;

    public StringSelection(String data) {
        this.data = data;
    }

    public DataFlavor[] getTransferDataFlavors() {
        return new DataFlavor[]{DataFlavor.stringFlavor};
    }

    public boolean isDataFlavorSupported(DataFlavor flavor) {
        return flavor == DataFlavor.stringFlavor;
    }

    public Object getTransferData(DataFlavor flavor) throws IOException {
        return data;
    }
}

File: src/jsMain/kotlin/awt/datatransfer/StringSelection.kt
package awt.datatransfer

actual class StringSelection actual constructor(private val data: String) : Transferable {
    private val flavors = arrayOf(STRING_FLAVOR)

    actual override fun getTransferDataFlavors(): Array<DataFlavor> = flavors
    actual override fun isDataFlavorSupported(flavor: DataFlavor): Boolean = flavor == STRING_FLAVOR
    actual override fun getTransferData(flavor: DataFlavor): Any {
        require(isDataFlavorSupported(flavor)) { "Unsupported flavor: ${flavor.getMimeType()}" }
        return data
    }
}
